test/blackbox-tests/test-cases/github2990/run.t

```diff
@@ -0,0 +1,3 @@
+Test for problem in #2990
+
+  $ dune runtest
```

Comment:
It seems to me that this test would pass with or without this PR. i.e. nothing is testing that `output.actual` is indeed not inferred as a target. I pushed a change to master to test this aspect.

Reply:
Ah, actually looking at the original bug report I see that without this PR we were indeed getting an error.